### ISSUE TITLE
rasterio maximum files limit

### ISSUE THREAD

**Issue by ptresson** [umr-amap/iamap#10]
Example on a large image

```
2024-09-05T11:52:58     CRITICAL    Traceback (most recent call last):
              File "rasterio/_base.pyx", line 310, in rasterio._base.DatasetBase.__init__
              File "rasterio/_base.pyx", line 221, in rasterio._base.open_dataset
              File "rasterio/_err.pyx", line 221, in rasterio._err.exc_wrap_pointer
             rasterio._err.CPLE_OpenFailedError: path/to/features/4704_0.tif: Trop de fichiers ouverts
```

Maybe merge and cleanup files at regular intervals ?

**Comment by ptresson** [comment]
may be fixed with 5277db655726b0b8ad3133a4fddd6d82ba976b06 but not tested

**Comment by ptresson** [comment]
5277db655726b0b8ad3133a4fddd6d82ba976b06 does not fix the issue but causes others.

may be related to https://github.com/rasterio/rasterio/issues/1636 or https://github.com/rasterio/rasterio/issues/1831
